cherrypy/lib/static.py

```diff
@@ -28,6 +29,21 @@ def _setup_mimetypes():
 _setup_mimetypes()
 
 
+def _utf8_content_disposition(disposition, file_name):
+    """Create HTTP header for downloading a file with UTF-8 filename.
+    See this and related answers:
+    https://stackoverflow.com/a/8996249/2173868
+    """
+    ascii_name = (
+        unicodedata.normalize('NFKD', file_name).
+        encode('ascii', errors='ignore').decode()
+    )
+    quoted_name = urllib.parse.quote(file_name)
+    header = u'{}; filename="{}"'.format(disposition, ascii_name)
```

Comment:
Appendix D says:
>    o  Include a "filename" parameter when US-ASCII ([US-ASCII]) is
>       sufficiently expressive.

Does this mean that `filename=` should be omitted if it's not expressive? Maybe when `not ascii_name.trim()`?

Reply:
I don`t think so, if you ommit `filename=`, there will be no filename specified for browser.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition

< __filename__
    Is followed by a string containing the original name of the file transmitted. The filename is always optional and must not be used blindly by the application: path information should be stripped, and conversion to the server file system rules should be done. This parameter provides mostly indicative information. When used in combination with Content-Disposition: attachment, it is used as the default filename for an eventual "Save As" dialog presented to the user.

Reply:
Okay, agreed.